### ISSUE TITLE
Logitech MX Master 3S

### ISSUE THREAD

**Issue by klfryrear** [libratbag/libratbag#1325]
I noticed that you have listed support for the Logitech MX Master 3. I recently picked up the updated version -- the MX Master 3S and it is not being recognized.

I would like to see support for the 3S, which after digging a little bit is different from the 3 in some way if not just product ID. Although, I suspect there may be more differences based on the receiver. It is using the "bolt" receiver, which I hadn't heard of until I bought this device.  

I tried just adding the device in the libratbag devices directory to no avail. I don't know if this device uses a different protocol than previous versions of the MX Master or not. 

I do know the device ID is 046d:c548. I would be happy to help with any other information that is needed.

**Comment by Raspberry-Pii** [comment]
I also have one! I would love to have support for the 3S!

**Comment by daniel-hartmann** [comment]
Did you try using bluetooth instead of the USB receiver?

I'm using the receiver here. And I tried using the receiver address and using the DeviceIndex as 1, but it didn't work.

`Bus 001 Device 006: ID 046d:c548 Logitech, Inc. USB Receiver`

[lsusb-046d:c548-verbose.txt](https://github.com/libratbag/libratbag/files/9300588/lsusb-046d.c548-verbose.txt)

**Comment by andolux** [comment]
Same, can't get it to recognize the MX Master 3S via receiver or bt. Also tried with HID++ 4.5 (hidpp45)

**Comment by NL-TCH** [comment]
Hey everyone see my pullrequest, tested and all: https://github.com/libratbag/libratbag/pull/1346

See also my pullrequest for pipersupport: https://github.com/libratbag/piper/pull/791
![image](https://user-images.githubusercontent.com/58732773/190860662-3f8bf153-51fa-47ca-af31-6a8b71138665.png)

**Comment by kodein1** [comment]
I also have one! I would love to have support for the 3S!

**Comment by kodein1** [comment]
Pls add Logitech MX Master 3s....

**Comment by NL-TCH** [comment]
the pull-request is now waiting to be verified (i re-requested a review)
when this is done, the piper-team will have to accept the piper-pull-request too

so at this point it is out of my hands and into the hands of the code maintainers/reviewers

**Comment by kodein1** [comment]
God, so this is why some repos don't accept pull requests from the master branch - GitHub doesn't let me push new commits there, but it let me update your branch to the upstream's master, which for some reason also closed the PR automatically.
(Or maybe you have the Allow edits and access to secrets by maintainers checkbox ticked off).

Either way, @NL-TCH, would you please make a new branch and make the same pull request from there?
Locally you would have to clone your repository, then just do git switch --create feat_logitech_mx_master_3s and git push (there may also be a way to do this from GitHub's Web UI, but I don't know how).
Edit: and don't forget to add the changes back, I forgot the changes got removed by my force-push...

**Comment by kodein1** [comment]
Mouse mx master 3s still nothing?

**Comment by staticssleever668** [comment]
@kodein1, for what it's worth, you can make the pull request yourself, just create a file under `data/devices/logitech-MX-Master-3S.device` with following contents:

```dosini
[Device]
Name=Logitech MX Master 3S
DeviceMatch=bluetooth:046d:b034
Driver=hidpp20
```

You can even do it through the GitHub's web UI.

---

Also a note: even when we get the pull request in, the support won't land to end users for a while, as libratbag doesn't see new releases often. If you want to use it now, you would have to create this file on the system yourself and put it to `/usr/share/libratbag`.

**Comment by staticssleever668** [comment]
Fixed by #1382.

**Comment by 991jo** [comment]
For me the MX Master 3S is working via bluetooth but not via USB.
